src/shell/Lexer_test.cpp

```diff
@@ -24,3 +24,24 @@ TEST_CASE("Lexer.basic")
     lexer.nextToken();
     CHECK(lexer.currentToken() == endo::Token::EndOfInput);
 }
+
+
+TEST_CASE("Lexer.brackets")
+{
+    auto lexer = endo::Lexer(std::make_unique<endo::StringSource>("$PATH ${PATH} ${{PATH}} "));
```

Comment:
Hm, I think we should also cover `"Hello $PATH."` and `"Hello ${PATH}."` to make it complete. :)